examples/step-56/step-56.cc

```diff
@@ -0,0 +1,492 @@
+                                      update_values              |
+                                      update_gradients           |
+                                      update_quadrature_points   |
+                                      update_JxW_values);
+    const unsigned int        dofs_per_cell = fe.dofs_per_cell;
+    const unsigned int        n_q_points    = quadrature_formula.size();
+    FullMatrix<double>        cell_matrix (dofs_per_cell, dofs_per_cell);
+    Vector<double>            cell_rhs (dofs_per_cell);
+    std::vector<double>       rhs_values(n_q_points);
+    std::vector<types::global_dof_index> local_dof_indices (dofs_per_cell);
+    const RightHandSide<spacedim> rhs;
+    for (typename DoFHandler<dim,spacedim>::active_cell_iterator
+         cell = dof_handler.begin_active(),
+         endc = dof_handler.end();
+         cell!=endc; ++cell)
+      if (cell->is_locally_owned())
+        {
+          cell_matrix = 0;
+          cell_rhs = 0;
+          fe_values.reinit (cell);
+          rhs.value_list (fe_values.get_quadrature_points(), rhs_values);
+          for (unsigned int i=0; i<dofs_per_cell; ++i)
+            for (unsigned int j=0; j<dofs_per_cell; ++j)
+              for (unsigned int q_point=0; q_point<n_q_points; ++q_point)
+                cell_matrix(i,j) += fe_values.shape_grad(i,q_point) *
+                                    fe_values.shape_grad(j,q_point) *
+                                    fe_values.JxW(q_point);
+          for (unsigned int i=0; i<dofs_per_cell; ++i)
+            for (unsigned int q_point=0; q_point<n_q_points; ++q_point)
+              cell_rhs(i) += fe_values.shape_value(i,q_point) *
+                             rhs_values[q_point]*
+                             fe_values.JxW(q_point);
+          cell->get_dof_indices (local_dof_indices);
+          constraints.distribute_local_to_global (cell_matrix,
+                                                  cell_rhs,
+                                                  local_dof_indices,
+                                                  system_matrix,
+                                                  system_rhs);
+        }
+    system_matrix.compress (VectorOperation::add);
+    system_rhs.compress (VectorOperation::add);
+  }
+  template <int spacedim>
+  void LaplaceBeltramiProblem<spacedim>::solve ()
+  {
+    TimerOutput::Scope t(computing_timer, "solve");
+    LA::MPI::Vector
+    completely_distributed_solution (locally_owned_dofs, mpi_communicator);
+    SolverControl solver_control (dof_handler.n_dofs(),
+                                  1e-7 * system_rhs.l2_norm());
+#ifdef USE_PETSC_LA
+    LA::SolverCG solver(solver_control, mpi_communicator);
+#else
+    LA::SolverCG solver(solver_control);
+#endif
+    LA::MPI::PreconditionAMG preconditioner;
+    LA::MPI::PreconditionAMG::AdditionalData data;
+#ifdef USE_PETSC_LA
+    data.symmetric_operator = true;
+#else
+    / * Trilinos defaults are good */
```

Comment:
If this is intended as a comment, then there should be no space between / and * at the beginning of the phrase...